java/ql/src/Likely Bugs/Termination/ConstantLoopCondition.ql

```diff
@@ -84,6 +84,9 @@ where
     not fa.getField().isFinal() and
     fa.getParent*() = cond
   ) and
-  not exists(ArrayAccess aa | aa.getParent*() = cond)
+  not exists(ArrayAccess aa | aa.getParent*() = cond) and
+  not exists(RValue use |
+    use.getParent*() = cond and use.getVariable().(LocalVariableDecl).isCaptured()
+  )
```

Comment:
I think we should revert these changes to the query and revert the changes in `Variable.qll`. The only thing we should keep in this PR is a test indicating the FPs. The query as written ought to resolve those FPs once we switch our SSA implementation to the shared library and add proper handling of Kotlin-captured variables.